### PR TITLE
Tlportal 193 - catch error when ESB produces unparsable response

### PR DESCRIPTION
**Pull request by dlhaines**

### CODE REVIEW

File: server/local/strings.yml [hunk 1]
```diff
@@ -13,5 +13,6 @@ strings: {
   lookAtOtherDashesNoCourses: "It looks like none of your classes are using web sites we know of.  Checkout the ",
   canvasUrl: "https://umich.instructure.com/",
   ctoolsUrl: "https://ctools.umich.edu/portal",
-  noCourses: "You seem to have no courses this term"
+  noCourses: "You seem to have no courses this term",
+  termFailure: "The service we depend on for term information is returning GARBAGE."
```
Comment (dlhaines) [inline server/local/strings.yml]: How about?

We are not finding your term information at the moment.  Please try again later.

File: UI/js/controllers.js [hunk 2]
```diff
@@ -126,35 +126,40 @@ dashboardApp.controller('termsController', ['Courses', 'Terms', '$rootScope', '$
...
   Terms.getTerms(termsUrl).then(function (data) {
-    // the ESB might return a sinble object rather than an array, turn it into an array
-    if (data.Result.length === undefined ){
-      data.Result = [].concat(data.Result);
+    if (data.failure) {
+      $scope.$parent.term  = $rootScope.lang.termFailure
     }
-    if (data.Result.length !==0)  {
+    else {
+      // the ESB might return a sinble object rather than an array, turn it into an array
```
Comment (zqian) [inline UI/js/controllers.js]: typo here: single

File: UI/js/controllers.js [hunk 3]
```diff
@@ -126,35 +126,40 @@ dashboardApp.controller('termsController', ['Courses', 'Terms', '$rootScope', '$
...
+      if (data.Result.length === undefined ){
+        data.Result = [].concat(data.Result);
+      }
+      if (data.Result.length !==0)  {
 
-      $scope.terms = data.Result;
-      $scope.$parent.term = data.Result[0].TermDescr;
-      $scope.$parent.termId = data.Result[0].TermCode;
+        $scope.terms = data.Result;
+        $scope.$parent.term = data.Result[0].TermDescr;
```
Comment (zqian) [inline UI/js/controllers.js]: so we get an array or construct an array with one element. Is it possible the array could have more than one element?

File: UI/js/controllers.js [hunk 4]
```diff
@@ -126,35 +126,40 @@ dashboardApp.controller('termsController', ['Courses', 'Terms', '$rootScope', '$
...
-    }  
+        Courses.getCourses(url).then(function (data) {
+          if (data.failure) {
+            $scope.courses.errors = data;
+            $scope.loading = false;
+          } else {
+            $scope.courses = data;
+            $scope.loading = false;
+          }
+          $('.colHeader small').append($('<span id="done" class="sr-only">' + $scope.courses.length + ' courses </span>'));
```
Comment (zqian) [inline UI/js/controllers.js]: should we append the term information into this line also to make the message more specific?

File: UI/js/controllers.js (hunk 2 above)
Comment (dlhaines) [inline UI/js/controllers.js]: Well that ought to be a word.  I'll fix it.

File: UI/js/controllers.js (hunk 4 above)
Comment (gsilver) [inline UI/js/controllers.js]: That is the problem - we do not have any term information at all.

File: UI/js/controllers.js (hunk 3 above)
Comment (gsilver) [inline UI/js/controllers.js]: So - the cascade goes something like this:
error >  display message  
no error > data is object > turn into an array with one object
no error > data is array > first term is the current term

**Pull request merged**